src/main/groovy/wooga/gradle/unity/utils/internal/UnityAssetFile.groovy

```diff
@@ -19,57 +19,147 @@ package wooga.gradle.unity.utils.internal
         validObject = isValid
     }
 
+    @Override
+    int hashCode() {
+        return content.hashCode()
+    }
+
+    @Override
+    boolean equals(Object obj) {
+        if (!obj) {
+            return false
+        }
+        if (obj instanceof UnityAssetFile) {
+            UnityAssetFile other = (UnityAssetFile)obj
+            return this.content == other.content
+        }
+        return false
+    }
+
+    Boolean isSerialized() {
+        assetFile != null
+    }
+
     Boolean isValid() {
         validObject
     }
 
+    File getAssetFile(){
+        assetFile
+    }
+
     Map<String, Object> getContent() {
         content
     }
 
-    static String stripUnityInstructions(String content) {
-        def lines = content.readLines()
-        lines.collect {
-            if (it.matches(/%TAG !u! tag:unity3d.com,.*:/)) {
-                return ""
-            }
+    boolean saveToFile() {
+        saveToFile(null)
+    }
 
-            def m = it =~ /(--- )!u!\d+( &\d+)/
-            if (m) {
-                return "${m[0][1]}${m[0][2]}"
-            }
+    // CONSIDER THIS API SOME MORE ....
+    boolean saveToFile(File assetFile,indent = 2){
 
-            return it
+        if (assetFile == null){
+            if (isSerialized()){
+                assetFile = this.assetFile
+            }
+            else{
+                return false
+            }
         }
-        .join("\n")
-    }
 
+        def options = new DumperOptions()
+        options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK)
+        options.indent = indent
+
+        Yaml parser = new Yaml(options)
+
+        def lines = []
 
+        // Append the yaml header and unity's instructions
+        lines.addAll(unityInstructions)
+
+        // Add the main map
+        def property = ["PlayerSettings" : content]
+        def contentAsYaml = parser.dump(property)
+        contentAsYaml = contentAsYaml.replaceAll(": null\n", ": \n")
+        lines.add(contentAsYaml)
+
+        // Add a newline at the end of the file
+        lines.add("\n")
+
+        def serialization = lines.join("\n")
+        assetFile.write(serialization)
+        return true
+    }
+
+//    static String stripUnityInstructions(String content) {
+//        def lines = content.readLines()
+//        lines.collect {
+//            if (it.matches(/%TAG !u! tag:unity3d.com,.*:/)) {
+//                return ""
+//            }
+//
+//            def m = it =~ /(--- )!u!\d+( &\d+)/
+//            if (m) {
+//                return "${m[0][1]}${m[0][2]}"
+//            }
+//
+//            return it
+//        }
+//        .join("\n")
+//    }
 }
+
+//class NullRepresenter extends Representer {
+//    NullRepresenter() {
+//        super();
+//        this.nullRepresenter = new RepresentNull();
+//    }
+//    private class RepresentNull implements Represent {
+//        Node representData(Object data) {
+//            return representScalar(Tag.NULL, "");
+//        }
+//    }
+//}
```

Comment:
Why is this no longer needed?

Reply:
Rather than needed, that was something I tried on this branch but didn't work out, will remove.